### PR TITLE
Fix towncrier error due to importlib_resources upgrade

### PR DESCRIPTION
**Pull request by sklam**
Pin towncrier and importlib_resources

### CODE REVIEW

**Comment by sklam** [comment]
Azure CI is failing due to `CondaHTTPError: HTTP 504 GATEWAY TIMEOUT`

File: docs/upcoming_changes/9069.bug_fix.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Fix towncrier
```
[the remark this answers is not in the document]
Comment (stuartarchibald) [reply docs/upcoming_changes/9069.bug_fix.rst:1]: Good point!

[the remark this answers is not in the document]
Comment (sklam) [reply docs/upcoming_changes/9069.bug_fix.rst:1]: Now that CI worked. I can go remove it

**Review by esc — APPROVED** [review]
LGTM towncrier works again.

**Review by esc — APPROVED** [review]
LGTM towncrier works again.

**Comment by esc** [comment]
I tried to re-run the GHAs for https://github.com/numba/numba/pull/7582 -- but it seems like towncrier is still "bust" somehow:

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.17/x64/bin/towncrier", line 8, in <module>
    sys.exit(cli())
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/click/core.py", line [14](https://github.com/numba/numba/actions/runs/5510954028/jobs/10094484938?pr=7582#step:5:15)34, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/hostedtoolcache/Python/3.9.[17](https://github.com/numba/numba/actions/runs/5510954028/jobs/10094484938?pr=7582#step:5:18)/x64/lib/python3.9/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/check.py", line 64, in _main
    __main(compare_with, directory, config)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/check.py", line 70, in __main
    base_directory, config = load_config_from_options(directory, config_path)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/_settings/load.py", line 79, in load_config_from_options
    config = load_config(base_directory)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/_settings/load.py", line 105, in load_config
    return load_config_from_file(directory, config_file)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/_settings/load.py", line 112, in load_config_from_file
    return parse_toml(directory, config)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/_settings/load.py", line 172, in parse_toml
    if not resources.is_resource(package, resource):
AttributeError: module 'importlib_resources' has no attribute 'is_resource'
Error: Process completed with exit code 1.
```

**Comment by esc** [comment]
> I tried to re-run the GHAs for #7582 -- but it seems like towncrier is still "bust" somehow:
> 
> ```
> Traceback (most recent call last):
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/bin/towncrier", line 8, in <module>
>     sys.exit(cli())
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
>     return self.main(*args, **kwargs)
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/click/core.py", line 1078, in main
>     rv = self.invoke(ctx)
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/click/core.py", line [14](https://github.com/numba/numba/actions/runs/5510954028/jobs/10094484938?pr=7582#step:5:15)34, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/opt/hostedtoolcache/Python/3.9.[17](https://github.com/numba/numba/actions/runs/5510954028/jobs/10094484938?pr=7582#step:5:18)/x64/lib/python3.9/site-packages/click/core.py", line 783, in invoke
>     return __callback(*args, **kwargs)
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/check.py", line 64, in _main
>     __main(compare_with, directory, config)
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/check.py", line 70, in __main
>     base_directory, config = load_config_from_options(directory, config_path)
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/_settings/load.py", line 79, in load_config_from_options
>     config = load_config(base_directory)
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/_settings/load.py", line 105, in load_config
>     return load_config_from_file(directory, config_file)
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/_settings/load.py", line 112, in load_config_from_file
>     return parse_toml(directory, config)
>   File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/towncrier/_settings/load.py", line 172, in parse_toml
>     if not resources.is_resource(package, resource):
> AttributeError: module 'importlib_resources' has no attribute 'is_resource'
> Error: Process completed with exit code 1.
> ```

Looking at:

https://github.com/numba/numba/actions/runs/5510954028/jobs/10094484938?pr=7582

seems like the unpinned variant is being used:

```
Run python3 -m pip install towncrier
```

**Pull request merged**